### ISSUE TITLE
Feature Request - Auto Unlock Chechbox

### ISSUE THREAD

**Issue by dave03284957** [danmunn/redmine_dmsf#646]
Is there a way we could get a feature that if checked, allows files to automatically unlock when whoever locked the file uploads a new version? Seem to be having an issue with users never unlocking their stuff.

**Comment by picman** [comment]
Yes, users with _Force file unlock_ permission are allowed to unlock documents locked by other users.

**Comment by dave03284957** [comment]
![image](https://cloud.githubusercontent.com/assets/24925141/22697244/25c1bbe2-ed0e-11e6-8fa9-ae6ae2fb9fab.png)

**Comment by dave03284957** [comment]
Does that diagram help explain what I'm looking for?

**Comment by picman** [comment]
Understood. At the moment the file remains locked even after a new revision upload.

Maybe in case of a new revision of a locked document we can alert the user with a message box whether the file should be released/unlocked while committing.

What do you think?

Because I don't want to change the present functionality. Rather make it optional.

**Comment by dave03284957** [comment]
I completely agree, making it optional makes far more sense. A message box would be great!

**Comment by dave03284957** [comment]
![image](https://cloud.githubusercontent.com/assets/24925141/22701335/9d75389c-ed1a-11e6-8076-4d238b7e7ca4.png)

**Comment by dave03284957** [comment]
My interpretation, I believe to be correct.
